pkg/kubelet/dockertools/exec.go

```diff
@@ -121,6 +126,14 @@ func (*NativeExecHandler) ExecInContainer(client DockerInterface, container *doc
 	if err != nil {
 		return err
 	}
+	if timeout > 0 {
+		defer time.AfterFunc(timeout, func() {
+			// FIXME: we should kill the process in the container,
+			// but I couldn't find anything in the Docker API docs
+			// on how to do it with the *Exec APIs.
```

Comment:
This seems unfortunate, but on the other hand it seems that there's an implicit timeout of 10s here (ticket frequency + count till 5 + break).